src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -901,4 +902,144 @@ final case class EServicesApiServiceImpl(
+                ra.multiAnswers.map(ma =>
+                  Json.obj("key" -> Json.fromString(ma.key), "values" -> Json.arr(ma.values.map(Json.fromString): _*))
+                ): _*
+              )
+            )
+          )
+          .getOrElse(Json.obj())
+      ).asJson
+    }
+
+    def checkRiskAnalysisForExport(eService: CatalogProcess.EService): Future[Unit] =
+      eService.mode match {
+        case CatalogProcess.EServiceMode.RECEIVE if eService.riskAnalysis.isEmpty =>
+          Future.failed(EServiceRiskAnalysisIsRequired(eService.id))
+        case _                                                                    => Future.unit
+      }
+
+    def verifyExportEligibility(descriptor: CatalogProcess.EServiceDescriptor): Future[Unit] =
+      descriptor.state match {
+        case CatalogProcess.EServiceDescriptorState.DRAFT =>
+          Future.failed(NotValidDescriptor(descriptor.id.toString, descriptor.state.toString))
+        case _                                            => Future.unit
+      }
+
+    def createZip(
+      folderName: String,
+      docs: Seq[(Array[Byte], String)],
+      interfaceFile: (Array[Byte], String),
+      config: Json
+    ): Array[Byte] = {
+      val bos    = new ByteArrayOutputStream()
+      val zipOut = new ZipOutputStream(bos)
+
+      docs.foreach { case (data, entryName) =>
+        zipOut.putNextEntry(new ZipEntry(s"$folderName/documents/$entryName"))
+        zipOut.write(data)
+        zipOut.closeEntry()
+      }
+
+      val (interfaceData, interfaceFileName) = interfaceFile
+      zipOut.putNextEntry(new ZipEntry(s"$folderName/$interfaceFileName"))
+      zipOut.write(interfaceData)
+      zipOut.closeEntry()
+
+      zipOut.putNextEntry(new ZipEntry(s"$folderName/configuration.json"))
+      zipOut.write(config.toString.getBytes("UTF-8"))
+      zipOut.closeEntry()
+
+      zipOut.close()
+      bos.toByteArray
+    }
+
+    val result: Future[(String, HttpEntity.Strict)] = for {
+      organizationId <- getOrganizationIdFutureUUID(contexts)
+      eServiceUuid   <- eserviceId.toFutureUUID
+      descriptorUuid <- descriptorId.toFutureUUID
+      eService       <- catalogProcessService.getEServiceById(eServiceUuid)
+      _              <- assertRequesterAllowed(eService.producerId)(organizationId)
+      _              <- checkRiskAnalysisForExport(eService)
+      descriptor     <- eService.descriptors
+        .find(_.id === descriptorUuid)
+        .toFuture(EServiceDescriptorNotFound(eService.id.toString, descriptorId))
+      _              <- verifyExportEligibility(descriptor)
+      interface      <- descriptor.interface.toFuture(MissingInterface(eServiceUuid, descriptorUuid))
+      interfaceFile  <- fileManager
+        .get(ApplicationConfiguration.eServiceDocumentsContainer)(interface.path)
+        .map(_.toByteArray)
+      _              <- FileManagerUtils
+        .verify(interfaceFile, interface.name, eService, isInterface = true)
+        .toFuture
```

Comment:
same, do we need this check during the export?
I would assume a correct configuration in the platform.

Anyway this is definitely needed during the import

Reply:
> same, do we need this check during the export? I would assume a correct configuration in the platform.
> 
> Anyway this is definitely needed during the import

[Same](https://github.com/pagopa/interop-be-backend-for-frontend/pull/277#discussion_r1627295923)